### ISSUE TITLE
Am I missing something?

### ISSUE THREAD

**Issue by cybernoctua** [guino/Merkury720#34]
I have downloaded the zip and the busy box outlined in the instructions. I then copy the contents of "mmc" onto the SD card. I overwrite the busybox file.

My confusion: The only way to get to it is via IP, the only way for it to get an IP is to connect it to wifi, the only way to connect it to wifi is give it the ssid and password of the network. It seems like thats what the "ppsFactoryTool.txt" file is used for, but that is not something that is copied to the SD card?!?

Basically what do I do with the rest of the downloaded files? Are they supposed to go on the SD card as well? 
"img" directory "jpeg" directory ".gitattributes" "ppsFactoryTool.txt" are technically not added to the SD based on how I am reading the instuctions.

help lol

**Comment by guino** [comment]
This is a listing from within the device (Linux) but you should have a good idea of what should be in the SD card:
```
/mnt/mmc01 # ls -la
-rwxr-xr-x    1 wagner   root        150444 Oct  7  2022 hostapd
-rwxr-xr-x    1 wagner   root           949 Aug  8  2022 log_parser.sh
-rwxr-xr-x    1 wagner   root           286 Sep 13  2021 hosts
-rwxr-xr-x    1 wagner   root          1066 Feb 22  2022 offline.sh
-rwxr-xr-x    1 wagner   root        257156 Jan 27  2022 mqtt_pub
-rwxr-xr-x    1 wagner   root         40056 Aug  6  2022 reredirect
-rwxr-xr-x    1 wagner   root            55 Feb  8  2021 ppsFactoryTool.txt
-rwxr-xr-x    1 wagner   root       2475072 Feb  8  2021 ppsapp
-rwxr-xr-x    1 wagner   root           263 Jan 27  2021 set
-rwxr-xr-x    1 wagner   root             5 Dec 31  1969 hack
-rwxr-xr-x    1 wagner   root            40 Jan  2  2021 passwd
-rwxr-xr-x    1 wagner   root          1327 Jan  2  2021 index.html
-rwxr-xr-x    1 wagner   root            18 Jan  2  2021 httpd.conf
-rwxr-xr-x    1 wagner   root           576 Feb 22  2022 custom.sh
drwxr-xr-x    2 wagner   root         16384 Feb  8  2021 bin
drwxr-xr-x    2 wagner   root         16384 Feb  8  2021 lib
drwxr-xr-x    7 wagner   root         16384 Feb  8  2021 home
drwxr-xr-x    2 wagner   root         16384 Jul 27  2021 cgi-bin
-rwxr-xr-x    1 wagner   root       1109128 Jan  3  2021 busybox
-rwxr-xr-x    1 wagner   root           161 Feb 14  2021 upload.html
-rwxr-xr-x    1 wagner   root           425 Feb  8  2021 initrun.sh
-rwxr-xr-x    1 wagner   root          7956 Mar  2  2021 jpeg-arm
-rwxr-xr-x    1 wagner   root           103 Jan  2  2021 ppsMmcTool.txt
drwxr-xr-x    3 wagner   root         16384 Jan  2  2021 SDT
-rwxr-xr-x    1 wagner   root            90 Jan  2  2021 env
drwxr-xr-x    3 wagner   root             0 Dec 31  1969 ..
drwxr-xr-x    8 wagner   root         16384 Jan 15 16:41 .

```

Please note step 2 of the instructions says to first copy what's __inside__ the mmc directory of the repository into the SD card's root **then** separately download the busybox file onto the SD card. Yes you should overwrite the busybox file with the one you separately downloaded (which should be over 1Mb in size as shown above).

You can get the IP of the device even without any files on the SD card -- just use the tyua/smartlife phone app to setup the device onto your network then find the IP of the device in your router. Once you have done that you should be able to use the ppsFactoryTool.txt (if required) to connect to the same network and the IP should most likely be the same (unless your router is doing weird stuff). You **must** initialize the device with the phone app before trying to use it for anything else, once it's initialized you don't need the phone app anymore.

Please notice the ppsFactoryTool.txt file is required to open port 80 on Firmware 2.7.6 and newer, 2.10.x and newer and 4.0.x, so if you have 2.7.3 or 2.9.6 you won't need it, but if you have 2.7.6, 2.10.x, 4.0.x you will need it. The notice about ppsFactoryTool.txt was added later (at the top of the repo information) and it does say that it needs to be saved onto the root of the SD card.

(The information about ppsFactoryTool was added later when we had to deal with newer firmware, so that's why it's on top before the steps at the bottom, so yeah, sorry it can be a bit confusing)

**Comment by linkian19** [comment]
@guino - Seeing this issue I hope you can clear up a quick question for me. In the ppsFactoryTool file when I modify ssid, my ssid has a space in it? Do I need a special character or just use the format: <word 1>(literal space)<word2>

I know you said to download into root (which I did) and edit, I assume just good old windows Notepad is fine to edit it with then?

Appreciate the help!

**Comment by guino** [comment]
@linkian19 as far as I know space should be fine, and yes you can can use notepad on that file, just don't mess with the `,,` in the file and don't touch the end of the line.

**Comment by linkian19** [comment]
@guino Awesome! Thank you! One more question: The password file, if I were to set a username and passwordHash, do I replace the user and passwordHash values or add them after the colons after those?

Was trying to telnet and couldn't find the right username and password combo so was going to set it.

Thanks again!

**Comment by guino** [comment]
@linkian19 you replace the `user` for your username i.e. john, and replace the `passwordHash` for the hash value generated on the site.

**Comment by linkian19** [comment]
@guino thank you for the clarification! Appreciate that!
